### PR TITLE
Master mda schema

### PR DESCRIPTION
**Pull request by mulkieran**

### CODE REVIEW

**Comment by mulkieran** [comment]
@agrover review?

File: docs/design/StratisSoftwareDesign.lyx [hunk 1]
```diff
@@ -3357,7 +3357,7 @@ size
 \begin_inset Text
 
 \begin_layout Plain Layout
-number
+integer
```
Comment (agrover) [inline docs/design/StratisSoftwareDesign.lyx]: The issue here is that JSON at least according to https://en.wikipedia.org/wiki/JSON doesn't distinguish between integers and floating point and just has a `number` type. Of course we want to limit valid values not just to integers but usually (always?) to nonnegative integers. Saying integer is more accurate, but it does mean we're not strictly using JSON types any more.

File: docs/design/StratisSoftwareDesign.lyx [hunk 2]
```diff
@@ -4408,5 +4416,396 @@ Journald interactions
 Nfs interactions
 \end_layout
 
+\begin_layout Part*
```
Comment (agrover) [inline docs/design/StratisSoftwareDesign.lyx]: We want the schema both in the design doc appendix and the separate file?

File: docs/design/StratisSoftwareDesign.lyx [hunk 3]
```diff
@@ -3243,8 +3243,8 @@ parameters of the thinpool device
 \end_layout
 
 \begin_layout Description
-block_devs: an object with 32-byte ASCII hex UUID keys and object values
- with the following structure:
+block_devs: an object with ASCII hex UUID keys and object values with the
+ following structure:
```
Comment (agrover) [inline docs/design/StratisSoftwareDesign.lyx]: I'd just be worried with this change it wouldn't be clear the length and format of the uuid we require. Maybe say 32 char?

File: docs/design/StratisSoftwareDesign.lyx (hunk 1 above)
Comment (mulkieran) [inline docs/design/StratisSoftwareDesign.lyx]: Yeah, I was using this recommended tutorial: https://spacetelescope.github.io/understanding-json-schema/, which believes in integer type.

File: docs/design/StratisSoftwareDesign.lyx (hunk 3 above)
Comment (mulkieran) [inline docs/design/StratisSoftwareDesign.lyx]: I think wherever else you mention UUID there is no statement of length.

File: docs/design/StratisSoftwareDesign.lyx (hunk 2 above)
Comment (mulkieran) [inline docs/design/StratisSoftwareDesign.lyx]: No, we want it in a separate file and included into the doc. I believe I figure out how to do that once...

**Comment by mulkieran** [comment]
Just need to solve the integer problem...

**Comment by mulkieran** [comment]
It is more convenient to merge this, we can follow up later on issue #31.

**Pull request merged**